nima_core/cognition/boundary_detector.py

```
@@ -139,8 +139,9 @@ def __init__(
         self.token_limit = token_limit
         self.message_limit = message_limit
         self.enable_llm_detection = enable_llm_detection
-        self._model = model
-        self._api_key = api_key or self._resolve_api_key()
+        llm_cfg = resolve_llm_config(model=model)
```

Ruff (0.15.10)

[error] Undefined name `resolve_llm_config`

(F821)